lxd/storage_lvm.go

```diff
@@ -763,7 +813,25 @@ func (s *storageLvm) ImageCreate(fingerprint string) error {
 		}
 	}()
 
-	err = s.tryMount(lvpath, tempLVMountPoint, "ext4", 0, "discard")
+	var fstype string
+	fstype, err = s.d.ConfigValueGet("storage.lvm_fstype")
+	if err != nil {
+		return fmt.Errorf("Error checking server config, err=%v", err)
+	}
+
+	// Generate a new xfs's UUID
+	if fstype == "xfs" {
+		output, err := exec.Command(
+			"xfs_admin",
+			"-U", "generate",
+			lvpath).CombinedOutput()
+		if err != nil {
+			// s.ContainerDelete(container)
+			return fmt.Errorf("Error generating new UUID: %v\noutput:'%s'", err, string(output))
+		}
+	}
```

Comment:
I don't see why we need to generate a new UUID here. We just got a new clean xfs filesystem created through mkfs.xfs, it's not a clone of any existing filesystem, so it should have a unique UUID.